File: solutions/lua/pythagorean-triplet/1/pythagorean-triplet.lua
return function(sum)
    local triplets = {}
    for a = 1, math.floor(sum / 3) do
        for b = a + 1, math.floor(sum / 2) do
            if (a * a + b * b == (sum - a - b) * (sum - a - b)) then
                table.insert(triplets, { a, b, (sum - a - b) })
            end
        end
    end
    return triplets
end

File: solutions/julia/pythagorean-triplet/1/pythagorean-triplet.jl
function pythagorean_triplets(n)
    res = []
    for c in n÷2-1:-1:floor(√2*n-n)+1
        d = √(c^2 - n^2 + 2n * c)
        d == floor(d) && push!(res, ((n - c - d) ÷ 2, (n - c + d) ÷ 2, c))
    end
    return res
end